src/DynamicsMethods/SurfaceHoppingMethods/surface_hopping.jl

```diff
@@ -123,6 +135,24 @@ function perform_rescaling!(
     return nothing
 end
 
+"""
+    frustrated_hop_invert_velocity!(
+        sim::AbstractSimulation{<:SurfaceHopping}, velocity, d
+    )
+
+Measures the component of velocity along the nonadiabatic coupling vector and inverts that component.
+"""
+function frustrated_hop_invert_velocity!(
+    sim::AbstractSimulation{<:SurfaceHopping}, velocity, d
+)
+    dn = d/norm(d)
```

Comment:
You could use normalize here instead:
```suggestion
    dn = LinearAlgebra.normalize(d)
```